docs/vaahflutter/core/api.md

```diff
@@ -0,0 +1,241 @@
+# API 
+
+::: warning Dependencies
+
+- [dio: ^4.0.6](https://pub.dev/packages/dio)
+
+- This panel uses EnvironmentConfig, thus depending on the env.dart file. It uses the base API URL from your EnviromentConfig.
+:::
+
+[[toc]]
+
+## Change Base URL
+
+To change the base URL change the variable in your [EnvironmentConfig](./environments).
+```dart
+'default': defaultConfig.copyWith(
+    apiUrl: 'YOUR_API_URL',
+),
+```
+The developer should not include the `/` at the end of the URL. e.g.
+```dart
+'default': defaultConfig.copyWith(
+    apiUrl: 'https://api.webreinvent.com',
+),
+```
+
+depending on `enableApiLogs` from [EnviromentConfig](./environments), `LogInterceptor` for dio can be enabled/ disabled.
+
+## Initialize Api class
+
+To use interceptors, Base Api Url, Cookie jar, etc, Api needs to be initialized first before use, so we initialize it using `Api.init()` in [BaseController](../app.md/#base-controller)
+
+```dart
+Api.init();
+```
+
+## Ajax method
+
+- ajax method automatically converts objects from `snake_case To lowerCamelCase` when `Receiving the response`.
+
+- ajax method automatically converts objects from `lowerCamelCase To snake_case` when `Sending the request`.
+
+### Parameters
+
+#### 1. URL
+- Parameter name: url
+- Required: yes
+- Type: `String`
+- Description: endpoint of a request
+- Note: Start the URL with `/`, because our base URL doesn't contain `/` at the end, [check here](#change-base-url).
+- Example:
+```dart
+Api.ajax(url: '/api/user');
+```
+
+#### 2. Callback
+- Parameter name: callback
+- Required: no
+- Type: `Future<void> Function(dynamic data, Response<dynamic>? res)?`
+- Description: callback function
+- Note: returns a null value when an error occurs, otherwise returns the value for the `data` key and whole response.
+- Example:
+```dart{3}
+Api.ajax(
+    url: '/api/users',
+    callback: (data, resp) async { Console.info('data', data); },
```

Comment:
@we-prajapati-c001 write 2 example 1. as inline `callback` as you have already written, 2. call back as new method eg. `sign` & `signAfter` as `callback`

Reply:
I've updated this and pushed a new commit.